### PR TITLE
Fix errors when lib built as C++

### PR DESCRIPTION
**Pull request by vrabaud**

### CODE REVIEW

File: src/reformat.c [hunk 1]
```diff
@@ -502,7 +502,7 @@ static avifBool avifCreateYUVToRGBLookUpTables(float ** unormFloatTableY, float
     const size_t cpCount = (size_t)1 << depth;
 
     assert(unormFloatTableY);
-    *unormFloatTableY = avifAlloc(cpCount * sizeof(**unormFloatTableY));
+    *unormFloatTableY = (float *)avifAlloc(cpCount * sizeof(**unormFloatTableY));
```
Comment (wantehchang) [inline src/reformat.c:505]: Optional: When there is a cast for the return value, I think using the `float` type in `sizeof` is more readable:

```
    *unormFloatTableY = (float *)avifAlloc(cpCount * sizeof(float));
```

Also at line 516.

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks!

File: src/reformat.c (hunk 1 above)
Comment (vrabaud) [reply src/reformat.c]: agreed. Done.

File: src/reformat.c [hunk 2]
```diff
@@ -502,7 +502,7 @@ static avifBool avifCreateYUVToRGBLookUpTables(float ** unormFloatTableY, float
     const size_t cpCount = (size_t)1 << depth;
 
     assert(unormFloatTableY);
-    *unormFloatTableY = (float *)avifAlloc(cpCount * sizeof(**unormFloatTableY));
+    *unormFloatTableY = (float *)avifAlloc(cpCount * sizeof(float));
```
Comment (wantehchang) [inline src/reformat.c:505]: Please make the same change to line 516. Thanks!

**Review by wantehchang — CHANGES_REQUESTED** [review]

File: src/reformat.c (hunk 2 above)
Comment (vrabaud) [reply src/reformat.c:505]: DOne

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**